cwms-data-api/src/test/java/cwms/cda/api/OpenApiDocTest.java

```diff
@@ -0,0 +1,486 @@
+        //We have a scope, so it's called from something like context.
+        return call.getScope().map(exp -> {
+            // First argument is the parameter name (usually a string literal or constant)
+            String paramName = parseParameterName(call.getArgument(0));
+
+            Class<?> paramClass = String.class;
+            if (call.getArguments().size() > 1) {
+                // Second argument is the class (e.g., Boolean.class, String.class)
+                ClassExpr argument = call.getArgument(1).asClassExpr();
+                paramClass = identifyClassFromExpression(unit, clazz, argument);
+            }
+            boolean used = true;
+            boolean nullHandled = true;
+            if (!required) {
+                //Check if null is handled via getOrDefault
+            }
+            return new OpenApiParamUsageInfo(new OpenApiParamInfo(paramName, required, paramClass), used, nullHandled);
+        }).orElseGet(() -> {
+            //It's calling a function, so most likely argument 0 is context, argument 1 is the identifier, and argument 2 is class.
+            String paramName = parseParameterName(call.getArgument(1));
+            Class<?> paramClass = String.class;
+            if (call.getArguments().size() > 2) {
+                ClassExpr argument = call.getArgument(2).asClassExpr();
+                paramClass = identifyClassFromExpression(unit, clazz, argument);
+            } else if (call.getNameAsString().endsWith("AsDouble")) {
+                paramClass = Double.class;
+            } else if (call.getNameAsString().endsWith("AsString")) {
+                paramClass = String.class;
+            }
+            boolean nullHandled = true;
+            return new OpenApiParamUsageInfo(new OpenApiParamInfo(paramName, required, paramClass), true, nullHandled);
+        });
+    }
+
+    private static @NotNull String parseParameterName(Expression arg) {
+        String value;
+        if (arg instanceof StringLiteralExpr) {
+            // Using a literal, which is problematic on its own, but we can get the value.
+            value = arg.asStringLiteralExpr().getValue();
+        } else if (arg instanceof FieldAccessExpr) {
+            // It's using a field accessor - this means it's calling Controllers.NAME for instance.
+            // This doesn't apply to when the field is statically imported though.
+            FieldAccessExpr fieldExp = arg.asFieldAccessExpr();
+            value = parseParameterName(fieldExp.getNameAsExpression());
+        } else if (arg instanceof NameExpr) {
+            // Assume it's coming from the Controllers class, as that's a normal location for constants
+            // This is effectively like...we have a name, but we're not sure where it's coming from and we're
+            // Not a string literal.
+            try {
+                Field field = Controllers.class.getField(arg.asNameExpr().getNameAsString());
```

Comment:
This is using Reflection to resolve the reference not JavaParser.  Thats interesting.   And its assuming the declaring class is always Controllers.   I think you could do something like this and know what the declaring class is.    Its usually going to be Controllers or the current class but it would be better to know than to guess.  

} else if (arg instanceof NameExpr) {
            NameExpr ne = (NameExpr) arg;
            ResolvedValueDeclaration resolved = ne.resolve();
            String declaringClass = resolved.asField().declaringType().getQualifiedName();
            String fieldName = arg.asNameExpr().getNameAsString();